packages/arb-tx-aggregator/txdb/txdb.go

```diff
@@ -192,27 +201,33 @@ func (txdb *TxDB) AddMessages(ctx context.Context, msgs []arbbridge.MessageDeliv
 			results = make([]resultInfo, 0)
 			txdb.callMut.Lock()
 			txdb.callMach = txdb.mach.Clone()
-			txdb.lastBlockId = block
+			txdb.lastBlockProcessed = block
 			txdb.callMut.Unlock()
 		}
 	}
-	if lastBlock != nil {
-		saveMach(txdb.mach, lastBlock, txdb.checkpointer)
-	}
 
 	txdb.callMut.Lock()
+	defer txdb.callMut.Unlock()
 	if txdb.callMach == nil || txdb.callMach.Hash() != txdb.mach.Hash() {
 		txdb.callMach = txdb.mach.Clone()
 	}
-	txdb.lastBlockId = finishedBlock
+	if len(messages) > 0 {
+		txdb.lastInboxSeq = messages[len(messages)-1].InboxSeqNum
+	}
+	txdb.lastBlockProcessed = finishedBlock
+	lastInboxSeq := txdb.lastInboxSeq
 	txdb.callMut.Unlock()
```

Comment:
This sits awkwardly with the defer of the same Unlock operation above.  I think you want one or the other, but not both.